packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -90,6 +93,86 @@ export async function getFileLink(
 	return fileLink;
 }
 
+async function getRequestInformation(
+	getToken: TokenFetcher<OdspResourceTokenFetchOptions>,
+	odspUrlParts: IOdspUrlParts,
+	logger: ITelemetryLoggerExt,
+	options: TokenFetchOptionsEx,
+	fileItem: FileItemLite,
+) {
+	const storageTokenFetcher = toInstrumentedOdspTokenFetcher(
+		logger,
+		odspUrlParts,
+		getToken,
+		true /* throwOnNullToken */,
+	);
+	const storageToken = await storageTokenFetcher(options, "GetFileLinkCore");
+	assert(
+		storageToken !== null,
+		0x2bb /* "Instrumented token fetcher with throwOnNullToken = true should never return null" */,
+	);
+
+	// IMPORTANT: In past we were using GetFileByUrl() API to get to the list item that was corresponding
+	// to the file. This was intentionally replaced with GetFileById() to solve the following issue:
+	// GetFileByUrl() uses webDavUrl to locate list item. This API does not work for Consumer scenarios
+	// where webDavUrl is constructed using legacy ODC format for backward compatibility reasons.
+	// GetFileByUrl() does not understand that format and thus fails. GetFileById() relies on file item
+	// unique guid (sharepointIds.listItemUniqueId) and it works uniformly across Consumer and Commercial.
+	const { url, headers } = getUrlAndHeadersWithAuth(
+		`${
+			odspUrlParts.siteUrl
+		}/_api/web/GetFileById(@a1)/ListItemAllFields/GetSharingInformation?@a1=guid${encodeURIComponent(
+			`'${fileItem.sharepointIds.listItemUniqueId}'`,
+		)}`,
+		storageToken,
+		true,
```

Comment:
Please add comments with arg names where it's not very clear what is the arg semantics
```ts
true, // forceAccessTokenViaAuthorizationHeader
```